### PR TITLE
Hide private links based on capabilities, refactor spans in sidebar t…

### PR DESCRIPTION
**Pull request by pascalwengerter**
## Description
This is a WIP PR, I'll push some more changes and add a changelog item later. Feel free to contribute suitable changes!

### CODE REVIEW

File: packages/web-app-files/src/components/Collaborators/Collaborator.vue [hunk 1]
```diff
@@ -140,8 +141,8 @@
         <div class="uk-flex uk-flex-nowrap uk-flex-middle">
           <oc-button
             v-if="$_editButtonVisible"
-            :aria-label="$gettext('Edit share')"
-            :uk-tooltip="$gettext('Edit share')"
+            :aria-label="$gettext(`Edit share with ${ collaborator.collaborator.displayName }`)"
```
Comment (kulmann) [inline packages/web-app-files/src/components/Collaborators/Collaborator.vue:144]: Unfortunately doesn't work like this. For inserting a variable into a translation, you need to use `gettextInterpolate`. And that's a 2-step-process. Translating the string (still with the variable) and then injecting the value into the translated string. See docs: https://github.com/Polyconseil/vue-gettext#interpolation-support-1 - I'd recommend to do this in a computed property.

Comment (pascalwengerter) [reply packages/web-app-files/src/components/Collaborators/Collaborator.vue:144]: Yeah I recall seeing this somewhere in the codebase, will push an update later today! Thanks ;)

Comment (kulmann) [reply packages/web-app-files/src/components/Collaborators/Collaborator.vue:144]: 👍

File: packages/web-app-files/src/components/PublicLinksSidebar/PrivateLinkItem.vue [hunk 2]
```diff
@@ -39,7 +37,7 @@ export default {
   }),
 
   computed: {
-    ...mapGetters('Files', ['highlightedFile']),
+    ...mapGetters('Files', ['highlightedFile'], ['capabilities']),
```
Comment (kulmann) [inline packages/web-app-files/src/components/PublicLinksSidebar/PrivateLinkItem.vue:40]: `mapGetters` expects two params: the namespace (can be omitted for root level) and **one** array or object. Please merge the two getter names into one array. ;-)

File: packages/web-runtime/src/components/Avatar.vue [hunk 3]
```diff
@@ -1,9 +1,9 @@
 <template>
-  <component :is="type" v-if="enabled">
+  <component :is="type" v-if="enabled" :aria-hidden="true">
     <oc-spinner
       v-if="loading"
       key="avatar-loading"
-      :aria-label="$gettext('Loading')"
+      :aria-label="$gettext('Loading user avatar')"
```
Comment (kulmann) [inline packages/web-runtime/src/components/Avatar.vue:6]: you can remove the aria-label entirely since the wrapper is aria-hidden

**Review by kulmann — APPROVED** [review]
LGTM 🥇

File: changelog/unreleased/enhancement-files-sidebar-accessibility [hunk 4]
```diff
@@ -2,12 +2,19 @@ Enhancement: Improve accessibility for the files sidebar
...
 - Transformed the favorite-star to a button-element
 - Adjusted aria-label of the favorite-star to describe what it does instead of its current state
 - Added a more descriptive close button label
 - Clicking outside of the sidebar now closes it
 - Removed the aria-label on the action buttons as they already include proper labels
 - Added a hint for screen readers if an action opens a new window/tab
+- Gave `role="presentation" to the collaborator avatar
+- Refactored `<span>` and `<div>` tags into `<p>` tags and unified translations a bit
+- Enhanced hints in the collaborator quick action buttons with collaborator name
+- Hide private links if the capability is not enabled
```
Comment (kulmann) [inline changelog/unreleased/enhancement-files-sidebar-accessibility:16]: IMO this is worth its own bugfix changelog

**Review by kulmann — APPROVED** [review]
LGTM 👍

**Pull request merged**